File: 1-two-sum/two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> Mp = new HashMap<>();
        int n = nums.length;

        for (int i = 0; i < n; i++) {
            Mp.put(nums[i], i);
        }

        for (int i = 0; i < n; i++) {
            int complement = target - nums[i];
            if (Mp.containsKey(complement) && Mp.get(complement) != i) {
                return new int[]{ i, Mp.get(complement)};
            }
        }

        return new int[]{}; 
    }
}

File: 1-two-sum/two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        for (int j = 0; j < nums.size(); j++) {
            int complete = target - nums[j];
            if (map.find(complete) != map.end()) {
                return {j, map[complete]};
            }
            map[nums[j]] = j;
        }
        throw invalid_argument("error");
    }
    
};